specutils/helcorr.py

```diff
@@ -0,0 +1,343 @@
+        sina = np.sin(a)
+        cosa = np.cos(a)
+        pertl += ccampm[k,0] * sina
+        pertld += ccampm[k,1] * cosa
+        pertp += ccampm[k,2] * cosa
+        pertpd -= ccampm[k,3] * sina
+
+    # Heliocentric motion of the Earth.
+    tl = forbel[1] + pertl
+    sinlm = np.sin(tl)
+    coslm = np.cos(tl)
+    sigma = cckm / (1.0 + pertp)
+    a = sigma * (ccmld + pertld)
+    b = sigma * pertpd
+    dxhd = dxhd + a * sinlm + b * coslm
+    dyhd = dyhd - a * coslm + b * sinlm
+    dzhd = -sigma * ccfdi * np.cos(forbel[2])
+
+    # Barycentric motion of the Earth.
+    dxbd = dxhd * dc1mme
+    dybd = dyhd * dc1mme
+    dzbd = dzhd * dc1mme
+    for k in range(0, 4):
+        plon = forbel[k + 3]
+        pomg = sorbel[k + 1]
+        pecc = sorbel[k + 9]
+        tl = (plon + 2.0 * pecc * np.sin(plon - pomg)) % (2 * np.pi)
+        dxbd += ccpamv[k] * (np.sin(tl) + pecc * np.sin(pomg))
+        dybd -= ccpamv[k] * (np.cos(tl) + pecc * np.cos(pomg))
+        dzbd -= ccpamv[k] * sorbel[k + 13] * np.cos(plon - sorbel[k + 5])
+
+    # Transition to mean equator of date.
+    dcosep = np.cos(deps)
+    dsinep = np.sin(deps)
+    dyahd = dcosep * dyhd - dsinep * dzhd
+    dzahd = dsinep * dyhd + dcosep * dzhd
+    dyabd = dcosep * dybd - dsinep * dzbd
+    dzabd = dsinep * dybd + dcosep * dzbd
+
+    dvelh = constants.au * (np.array([dxhd, dyahd, dzahd])) / u.second
+    dvelb = constants.au * (np.array([dxbd, dyabd, dzabd])) / u.second
+    return (dvelh, dvelb)
+
+
+# NOTE:
+# We may want to change the syntax input for helcorr so that it accepts a single
+# sky coordinate instead of ra/dec.
+# Similarly lon/lat/alt/jd could be replaced with a single astropy.units.Time
+# class.
+
+def helcorr(lon, lat, alt, ra, dec, mjd):
+    """
+    Calculate the heliocentric radial velocity corrections for an astronomical 
+    source.
+
+    Parameters
+    ----------
+    lon : `~astropy.coordinates.Longitude` or float
+        Earth longitude of the observatory (western direction is positive). Can
+        be anything that initialises an `~astropy.coordinates.Angle` object
+        (if float, in degrees).
+    lat : `~astropy.coordinates.Latitude` or float
+        Earth latitude of observatory. Can be anything that initialises an
+        `~astropy.coordinates.Latitude` object (if float, in degrees).
+    alt : `~astropy.units.Quantity` or float
+        Altitude of the observatory (if float, in meters).
+    ra : `~astropy.coordinates.Angle` or float
+        Right ascension of the object for epoch J2000 (if float, in degrees).
+    dec : `~astropy.coordinates.Angle` or float
+        Declination of the object for epoch J2000 (if float, in degrees).
+    mjd : float
+        The modified Julian date for the middle of exposure.
+
+    Returns
+    -------
+    helcorr : `~astropy.units.Quantity`
+        The heliocentric velocity correction.
+    """
+
+    if not isinstance(lon, coord.Longitude):
```

Comment:
I would suggest to require that these things are quantities or directly coordinates, to make sure that users definitely know what they are inputting. Maybe using an example section in the input:

```python

Example
-------

helcor(20*u.deg, 155*u.deg, ....)```